platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -664,12 +680,18 @@ OtaPalImageState_t otaPal_GetPlatformImageState( OtaFileContext_t * const C )
     FILE * pPlatformImageState = NULL;
     OtaImageState_t eSavedAgentState = OtaImageStateUnknown;
     OtaPalImageState_t ePalState = OtaPalImageStateUnknown;
+    OtaPalPathGenStatus_t status = OtaPalFileGenSuccess;
     char imageStateFile[ OTA_FILE_PATH_LENGTH_MAX ] = { 0 };
 
     ( void ) C;
 
     /* Get file path for the image state file. */
-    getFilePathFromCWD( imageStateFile, OTA_PLATFORM_IMAGE_STATE_FILE );
+    status = getFilePathFromCWD( imageStateFile, OTA_PLATFORM_IMAGE_STATE_FILE );
+
+    if( status != OtaPalFileGenSuccess )
+    {
+        LogError( ( "Could not generate the absolute path for the file" ) );
+    }
 
     /* POSIX port using standard library */
     /* coverity[misra_c_2012_rule_21_6_violation] */
```

Comment:
If `status != OtaPalFileGenSuccess` on line 691, you're opening an invalid filename here.  You should not continue processing in this case.

Reply:
Refactored in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1487/commits/a3d852db7cfbb536eddf4de696d8ec0f070a7702